2024_wsb/Src/encoders.c

```diff
@@ -1,6 +1,18 @@
-#include "stm32f4xx_hal.h"
+#include "tim.h"
+
+HAL_StatusTypeDef hallEffect_init(void) {
+    if (HAL_TIM_Base_Init(&htim5) != HAL_OK) {
```

Comment:
Don't use the htim5 directly like this, add it to the bsp.h file as a #define and use that, that way it is easily reconfigurable.
`
#define HALL_EFFECT_TIM_HANDLE htim5
`